### PR TITLE
Update/kws 2645 add video player to feed

### PR DESCRIPTION
**Pull request by devgabrielcoman**
## JIRA
[KWS-2645](https://jira.superawesome.tv/browse/KWS-2645)

## DESCRIPTION
Added the "live video player" - `RichVideoView` - to the `SingleFeedItemView`, which renders a single `Feed Item`, but with all live details (images, rich media (?) and video, if that's the case). 

Looks good so far, but:
- it doesn't go **fullscreen** at the moment
- when it encounters and error it doesn't do anything but display the error

## SCREENSHOTS

A view of the Video Player in action:
screenshot_20180613-172954

And in the Feed, where it's just an image:
screenshot_20180613-173020

### CODE REVIEW

File: popjamui/src/main/java/tv/superawesome/pj/ui/common/views/images/RoundedCornersTransformation.kt [hunk 1]
```diff
@@ -14,11 +14,12 @@ import java.util.*
  * Created by javiergonzalezcabezas on 2/4/16.
  */
 class RoundedCornersTransformation(context: Context,
+                                   private val source: String,
                                    private val cornerRadius: Int,
                                    private val borderWidth: Int,
                                    private val borderColor: Int? = null): BitmapTransformation(context) {
     override fun getId(): String {
-        return UUID.randomUUID().toString()
+        return source // UUID.randomUUID().toString()
```
Comment (me2k) [inline popjamui/src/main/java/tv/superawesome/pj/ui/common/views/images/RoundedCornersTransformation.kt]: remove the comment!!

File: popjamui/src/main/java/tv/superawesome/pj/ui/common/views/video/mediaplayer/IRichMediaPlayer.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import android.media.MediaPlayer
+
+/**
+ * Created by gabriel.coman on 13/06/2018.
+ */
+interface IRichMediaPlayer:
+        MediaPlayer.OnPreparedListener,
+        MediaPlayer.OnErrorListener,
+        MediaPlayer.OnCompletionListener {
+
```
Comment (me2k) [inline popjamui/src/main/java/tv/superawesome/pj/ui/common/views/video/mediaplayer/IRichMediaPlayer.kt]: 2 new lines here... make it 1 or less

File: popjamui/src/main/java/tv/superawesome/pj/ui/feed/views/FeedItemView.kt [hunk 3]
```diff
@@ -54,10 +61,29 @@ constructor(ctx: Context, attrs: AttributeSet? = null, style: Int = 0): Relative
...
-        feedItemImage?.loadImage(viewModel.feedImageUrl)
+
+        if (viewModel.isVideoItem) {
+            feedItemVideo?.visibility = View.VISIBLE
+            feedItemImage?.visibility = View.GONE
+            viewModel.videoUrl?.let {
+                feedItemVideo?.play(it)
+            }
+        }
+        else {
```
Comment (me2k) [inline popjamui/src/main/java/tv/superawesome/pj/ui/feed/views/FeedItemView.kt]: move this else up to the previous line

File: popjamui/src/main/java/tv/superawesome/pj/ui/common/views/images/RoundedCornersTransformation.kt (hunk 1 above)
Comment (usmaan-sa) [reply popjamui/src/main/java/tv/superawesome/pj/ui/common/views/images/RoundedCornersTransformation.kt]: I second this.

File: popjamui/src/main/java/tv/superawesome/pj/ui/common/views/adapter/AbstractAdapter.kt [hunk 4]
```diff
@@ -43,6 +44,7 @@ abstract class AbstractAdapter<VM: IRepeatingViewModel, VH: RecyclerView.ViewHol
         val context = parent.context
         return when (viewType) {
             R.layout.vanilla_skin_standard_header -> {
+                Log.d("SuperAwesome-Video", "I'm in the header!")
```
Comment (GuilhermeMota93) [inline popjamui/src/main/java/tv/superawesome/pj/ui/common/views/adapter/AbstractAdapter.kt]: Are we leaving the logs in?

File: popjamui/src/main/java/tv/superawesome/pj/ui/common/views/adapter/AbstractAdapter.kt [hunk 5]
```diff
@@ -53,6 +55,10 @@ abstract class AbstractAdapter<VM: IRepeatingViewModel, VH: RecyclerView.ViewHol
 
     override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
 
+        if (position == 0) {
+            Log.d("SuperAwesome-Video", "Binding at position 0")
```
Comment (GuilhermeMota93) [inline popjamui/src/main/java/tv/superawesome/pj/ui/common/views/adapter/AbstractAdapter.kt]: Again, are we keeping logs?

File: popjamui/src/main/java/tv/superawesome/pj/ui/common/views/seekbar/RichSeekBar.kt [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        (0..array.indexCount - 1)
+                .map { array.getIndex(it) }
+                .forEach {
+                    when (it) {
+                        R.styleable.RichSeekBar_barThumbColor -> {
+                            val color = array.getColor(it, Color.WHITE)
+                            thumb.setColorFilter(color, PorterDuff.Mode.SRC_IN);
+                        }
+                        R.styleable.RichSeekBar_barProgressColor -> {
+                            val color = array.getColor(it, Color.WHITE)
```
Comment (GuilhermeMota93) [inline popjamui/src/main/java/tv/superawesome/pj/ui/common/views/seekbar/RichSeekBar.kt]: Default color to be WHITE? Or should we actually have a default one?

File: popjamui/src/main/java/tv/superawesome/pj/ui/common/views/video/mediaplayer/RichMediaPlayer.kt [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    private fun readMediaError(code: Int): String = when (code) {
+        MediaPlayer.MEDIA_ERROR_MALFORMED -> "Media URL is invalid"
+        MediaPlayer.MEDIA_ERROR_SERVER_DIED -> "Server died"
+        MediaPlayer.MEDIA_ERROR_TIMED_OUT -> "Server timed out"
+        MediaPlayer.MEDIA_ERROR_IO -> "I/O Error"
+        MediaPlayer.MEDIA_ERROR_NOT_VALID_FOR_PROGRESSIVE_PLAYBACK -> "Media not valid for progressive playback"
+        MediaPlayer.MEDIA_ERROR_UNSUPPORTED -> "Media unsupported"
+        MediaPlayer.MEDIA_ERROR_UNKNOWN -> "Unknown error"
```
Comment (GuilhermeMota93) [inline popjamui/src/main/java/tv/superawesome/pj/ui/common/views/video/mediaplayer/RichMediaPlayer.kt]: Get all the Strings in a string file maybe?

**Pull request merged**